Chetan/teleport

Madrobot9182 opened the pull request:
- Added backwards transitions with a gradient effect to hopefully direct the player better
- Slightly changed the tile map to accommodate this
- Minor logic fix for intro sequence (teleport directly after dialog)

shellbertt commented:
* we don't need a teleport back to the intro room in the first campfire
* the teleports to campfire disappears after we go through it so if we go through and return we are trapped
* need to place teleports and landing spots a bit better

Madrobot9182 commented:
> * we don't need a teleport back to the intro room in the first campfire
> 
>     * the teleports to campfire disappears after we go through it so if we go through and return we are trapped
> 
>     * need to place teleports and landing spots a bit better

- Removed the transitions such that once you reach the city you can't go back to fireplace 1 (didn't make sense to go that far back anyways)
- Not sure what targets could be done better but I went through all of them again and moved some closer to the exit spots

chetanty commented:
right now when i go to the second camp and come back to city2, there is no more x so i cannot progress.
remove shaders in the crashed bus camp, it looks odd and we cant really go back from there.
we can add share in the second camp as well.

Madrobot9182 commented:
> right now when i go to the second camp and come back to city2, there is no more x so i cannot progress. 
Looks like the fireplace script disables the object it's on (ie the transition object), with no way of re-enabling later)
> remove shaders in the crashed bus camp, it looks odd and we cant really go back from there. 
I think that's an old version. Could you pull again and check
> we can add share in the second camp as well.
Top and right side already has the shade I'll add it to all sides. I think it'll look good

chetanty reviewed (approved):
looks good, ready to merge

shellbertt commented:
merging, need to fix:

teleport getting deleted when going back
maybe don't have bed teleport you anymore but guide the player to move on after sleeping
don't let them walk out before doing the campfire initially
